Tests from fresh DB fail

When running pytest with `--create-db --migrations -n auto` some tests fail. They pass again when running with just `-n auto` immediately after.

```
__________________________________ test_terminal_states_have_no_transitions ___________________________________
[gw8] darwin -- Python 3.12.7 /Users/tom/code/gyrinx/gyrinx/.venv/bin/python3

self = <django.db.backends.utils.CursorWrapper object at 0x11c89dac0>
sql = 'INSERT INTO "core_statemachinetestmodel" ("id", "created", "modified", "name", "status") VALUES (%s, %s, %s, %s, %s)'
params = (UUID('244bfec3-40e0-40e1-b037-30688c15d53b'), datetime.datetime(2026, 2, 28, 19, 50, 29, 160074, tzinfo=datetime.timezone.utc), datetime.datetime(2026, 2, 28, 19, 50, 29, 160080, tzinfo=datetime.timezone.utc), 'test', 'DONE')
ignored_wrapper_args = (False, {'connection': <DatabaseWrapper vendor='postgresql' alias='default'>, 'cursor': <django.db.backends.utils.CursorWrapper object at 0x11c89dac0>})

    def _execute(self, sql, params, *ignored_wrapper_args):
        # Raise a warning during app initialization (stored_app_configs is only
        # ever set during testing).
        if not apps.ready and not apps.stored_app_configs:
            warnings.warn(self.APPS_NOT_READY_WARNING_MSG, category=RuntimeWarning)
        self.db.validate_no_broken_transaction()
        with self.db.wrap_database_errors:
            if params is None:
                # params default might be backend specific.
                return self.cursor.execute(sql)
            else:
>               return self.cursor.execute(sql, params)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E               psycopg2.errors.UndefinedTable: relation "core_statemachinetestmodel" does not exist
E               LINE 1: INSERT INTO "core_statemachinetestmodel" ("id", "created", "...
E                                   ^

.venv/lib/python3.12/site-packages/django/db/backends/utils.py:105: UndefinedTable

The above exception was the direct cause of the following exception:

    @pytest.mark.django_db
    def test_terminal_states_have_no_transitions():
        """DONE and FAILED have no outbound transitions."""
>       obj = StateMachineTestModel.objects.create(status="DONE")
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

gyrinx/core/tests/test_state_machine.py:144:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
.venv/lib/python3.12/site-packages/django/db/models/manager.py:87: in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/django/db/models/query.py:669: in create
    obj.save(force_insert=True, using=self.db)
.venv/lib/python3.12/site-packages/django/db/models/base.py:874: in save
    self.save_base(
.venv/lib/python3.12/site-packages/django/db/models/base.py:966: in save_base
    updated = self._save_table(
.venv/lib/python3.12/site-packages/django/db/models/base.py:1169: in _save_table
    results = self._do_insert(
.venv/lib/python3.12/site-packages/django/db/models/base.py:1220: in _do_insert
    return manager._insert(
.venv/lib/python3.12/site-packages/django/db/models/manager.py:87: in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/django/db/models/query.py:1918: in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/django/db/models/sql/compiler.py:1925: in execute_sql
    cursor.execute(sql, params)
.venv/lib/python3.12/site-packages/django/db/backends/utils.py:79: in execute
    return self._execute_with_wrappers(
.venv/lib/python3.12/site-packages/django/db/backends/utils.py:92: in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/django/db/backends/utils.py:100: in _execute
    with self.db.wrap_database_errors:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
.venv/lib/python3.12/site-packages/django/db/utils.py:94: in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <django.db.backends.utils.CursorWrapper object at 0x11c89dac0>
sql = 'INSERT INTO "core_statemachinetestmodel" ("id", "created", "modified", "name", "status") VALUES (%s, %s, %s, %s, %s)'
params = (UUID('244bfec3-40e0-40e1-b037-30688c15d53b'), datetime.datetime(2026, 2, 28, 19, 50, 29, 160074, tzinfo=datetime.timezone.utc), datetime.datetime(2026, 2, 28, 19, 50, 29, 160080, tzinfo=datetime.timezone.utc), 'test', 'DONE')
ignored_wrapper_args = (False, {'connection': <DatabaseWrapper vendor='postgresql' alias='default'>, 'cursor': <django.db.backends.utils.CursorWrapper object at 0x11c89dac0>})

    def _execute(self, sql, params, *ignored_wrapper_args):
        # Raise a warning during app initialization (stored_app_configs is only
        # ever set during testing).
        if not apps.ready and not apps.stored_app_configs:
            warnings.warn(self.APPS_NOT_READY_WARNING_MSG, category=RuntimeWarning)
        self.db.validate_no_broken_transaction()
        with self.db.wrap_database_errors:
            if params is None:
                # params default might be backend specific.
                return self.cursor.execute(sql)
            else:
>               return self.cursor.execute(sql, params)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E               django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
E               LINE 1: INSERT INTO "core_statemachinetestmodel" ("id", "created", "...
E                                   ^

.venv/lib/python3.12/site-packages/django/db/backends/utils.py:105: ProgrammingError
=========================================== short test summary info ===========================================
FAILED gyrinx/core/tests/test_state_machine.py::test_is_terminal_property - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_to_valid_state_succeeds - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_to_invalid_state_raises - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_creates_record - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_records_from_status - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_records_to_status - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_records_timestamp - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_accepts_metadata - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_multiple_transitions_create_multiple_records - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_history_returns_all_transitions - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_history_first_returns_latest - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_history_empty_when_no_transitions - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_display_property - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_str - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_links_to_instance - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_with_save_false - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_transition_metadata_defaults_to_empty_dict - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_initial_state_set_on_create - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_explicit_status_preserved - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_can_transition_to_valid_state - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_cannot_transition_to_invalid_state - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_cannot_transition_to_same_state - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_get_valid_transitions_from_pending - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_get_valid_transitions_from_running - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
FAILED gyrinx/core/tests/test_state_machine.py::test_terminal_states_have_no_transitions - django.db.utils.ProgrammingError: relation "core_statemachinetestmodel" does not exist
=========================== 25 failed, 2072 passed, 13 skipped in 418.40s (0:06:58) ===========================
```

--- a/gyrinx/core/tests/test_state_machine.py
+++ b/gyrinx/core/tests/test_state_machine.py
@@ -3,7 +3,7 @@ Tests for the StateMachine descriptor and per-model transition tables.
 """
 
 import pytest
-from django.db import models
+from django.db import connection, models
 
 from gyrinx.core.models.state_machine import (
     InvalidStateTransition,
@@ -34,6 +34,24 @@ class StateMachineTestModel(Base):
 
     class Meta:
         app_label = "core"
+
+
+@pytest.fixture(scope="session", autouse=True)
+def _ensure_state_machine_test_tables(django_db_setup, django_db_blocker):
+    """Create tables for test-only models when running with --migrations.
+
+    With --nomigrations (the default), syncdb creates tables for all discovered
+    models including test models. With --migrations, only migrated tables are
+    created, so this fixture fills the gap.
+    """
+    with django_db_blocker.unblock():
+        table_names = connection.introspection.table_names()
+        if "core_statemachinetestmodel" not in table_names:
+            with connection.schema_editor() as schema_editor:
+                schema_editor.create_model(StateMachineTestModel)
+                schema_editor.create_model(
+                    StateMachineTestModel.states.transition_model
+                )
 
 
 # =============================================================================

